painting/src/main/java/com/ataulm/artcollector/painting/ui/PaintingActivity.kt

```diff
@@ -61,19 +59,19 @@ class PaintingActivity : AppCompatActivity() {
 
     private fun Intent.loadImageIfAvailable() {
         imageUrl()?.let {
-            paintingImageViewTarget = glideRequestManager
-                    .load(it)
+            glideRequestManager.load(it)
                     .listener(startTransitionRequestListener)
                     .into(imageView)
         }
     }
 
     private fun Painting.loadImageIfDifferent() {
-        paintingImageViewTarget?.request?.clear()
-        paintingImageViewTarget = glideRequestManager
-                .load(imageUrl)
-                .listener(startTransitionRequestListener)
-                .into(imageView)
+        if (imageUrl != intent.imageUrl()) {
+            glideRequestManager.clear(imageView)
```

Comment:
can clear the target just by passing the imageview. internally it uses view tags, so we can't use them without an ID too (not sure why _it_ doesn't use an ID..) but we don't actually use them so it doesn't matter right now 😬